src/components/navigation/Tab.tsx

```diff
@@ -0,0 +1,88 @@
+import classnames from 'classnames';
+import type { JSX } from 'preact';
+
+import type { IconComponent, PresentationalProps } from '../../types';
+import { downcastRef } from '../../util/typing';
+
+import ButtonBase from '../input/ButtonBase';
+
+type ComponentProps = {
+  icon?: IconComponent;
+  /**
+   * Text string representing the content of the tab when selected. The tab
+   * button will be sized to accommodate this string in bold text. This can
+   * prevent tab jiggle.
+   */
+  textContent?: string;
```

Comment:
I suppose we could try to simplify the API by determining the `textContent` automatically, eg. in a `useLayoutEffect` that reads `textContent` after render, although that would have a limitation of not being able to account for expected text changes (eg. if counts were not available and become available later), and be a bit more expensive.

Reply:
Yeah, I'm going to continue to hem and haw on this some. I don't plan to immediately cut a release of the package, so something intelligent might occur to me in the next few days.